app/src/main/java/org/mozilla/focus/session/SessionCallbackProxy.java

```diff
@@ -124,4 +142,19 @@ public void onExitFullScreen() {
         delegate.onExitFullScreen();
     }
 
+    private void recordEndOfLoad() {
+        if (urlLoading != null && session.getUrl().getValue().equals(urlLoading)) {
+            Log.i(LOG_TAG, "Loaded page at " + session.getUrl().getValue());
+            final long endTime = SystemClock.elapsedRealtime();
+            Log.i(LOG_TAG, "zerdatime " + endTime + " - page load stop");
+            final long elapsedLoad = endTime - startLoadTime;
+            Log.i(LOG_TAG, elapsedLoad + " - elapsed load");
+            // Even internal pages take longer than 30 ms to load, let's not send any loads faster than this
+            if (elapsedLoad > MIN_LOAD_TIME && !UrlUtils.isLocalizedContent(urlLoading)) {
+                Log.i(LOG_TAG, "Sent load to histogram");
+                TelemetryWrapper.addLoadToHistogram(elapsedLoad);
+            }
+        }
+    }
+
```

Comment:
It feels a bit dirty stuffing all this into `SessionCallbackProxy`- It's primary use is to update `Session` from the IWebView callbacks.
Can we move this into a separate class (like before) and make it subscribe to "loading" and "URL changed" events?